tsdb/db.go

```diff
@@ -597,6 +597,85 @@ func (db *DBReadOnly) Blocks() ([]BlockReader, error) {
 	return blockReaders, nil
 }
 
+// LastBlockID returns the BlockID of latest block.
+func (db *DBReadOnly) LastBlockID(logger log.Logger) (string, error) {
+	select {
+	case <-db.closed:
+		return "", ErrClosed
+	default:
+	}
+	latestDirName, err := db.lastBlockDirName(logger)
+	if err != nil {
+		return "", err
+	}
+	// Open the latest block and get its ID
+	block, err := db.Block(logger, latestDirName)
+	if err != nil {
+		return "", err
+	}
+	return block.Meta().ULID.String(), nil
+}
+
+func (db *DBReadOnly) lastBlockDirName(logger log.Logger) (string, error) {
+	latestDirName := ""
```

Comment:
Golang Convention Tip: The variable declaration should be as close as it can be to it's usage. Here, it has been defined in the beginning of the function block but used at almost the end of the block. It should be declared at around line 635 where the loop starts which utilises this variable.

Reply:
Okay, got it